nikola/plugins/task/page_index.py

```diff
@@ -64,6 +64,8 @@ def is_enabled(self, lang=None):
     def classify(self, post, lang):
         """Classify the given post for the given language."""
         destpath = post.destination_path(lang, sep='/')
+        if post._has_pretty_url(lang):
+            destpath = destpath.rstrip('/index.html')
```

Comment:
Actually, `rstrip` is not a good idea here at all: it removes all letters from the right of the string which are *contained* in `/index.html`. So also `/end/html/nix.html` would be removed, for example.